System/Random.hs

```diff
@@ -129,6 +210,10 @@ mutableByteArrayContentsCompat = mutableByteArrayContents
 -- | The class 'RandomGen' provides a common interface to random number
 -- generators.
 --
+-- N.B. Using 'next' is inefficient as all operations go via
+-- 'Integer'. See
+-- [here](https://alexey.kuleshevi.ch/blog/2019/12/21/random-benchmarks)
+-- for more details.
```

Comment:
I think `next` should be deprecated: https://github.com/idontgetoutmuch/random/pull/24

Reply:
I have marked it as deprecated and will write some words as to what users should do (presumably sprinkle a few `fromInteger` calls around.)

Reply:
#24 contains a default implementation of the deprecated `next`. That way, PRNG authors don't need to implement or care about `next` at all, and it will be easier to remove in the future.

Reply:
I cherry picked https://github.com/idontgetoutmuch/random/pull/24